homeassistant/components/workday/config_flow.py

```diff
@@ -0,0 +1,305 @@
+        vol.Optional(CONF_OFFSET, default=DEFAULT_OFFSET): NumberSelector(
+            NumberSelectorConfig(min=-10, max=10, step=1, mode=NumberSelectorMode.BOX)
+        ),
+        vol.Optional(CONF_WORKDAYS, default=DEFAULT_WORKDAYS): SelectSelector(
+            SelectSelectorConfig(
+                options=ALLOWED_DAYS, multiple=True, mode=SelectSelectorMode.DROPDOWN
+            )
+        ),
+        vol.Optional(CONF_ADD_HOLIDAYS, default=[]): SelectSelector(
+            SelectSelectorConfig(
+                options=[],
+                multiple=True,
+                custom_value=True,
+                mode=SelectSelectorMode.DROPDOWN,
+            )
+        ),
+        vol.Optional(CONF_REMOVE_HOLIDAYS, default=[]): SelectSelector(
+            SelectSelectorConfig(
+                options=[],
+                multiple=True,
+                custom_value=True,
+                mode=SelectSelectorMode.DROPDOWN,
+            )
+        ),
+    }
+)
+
+
+class WorkdayConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Workday integration."""
+
+    VERSION = 1
+
+    data: dict[str, Any] = {}
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> WorkdayOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return WorkdayOptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import a configuration from config.yaml."""
+
+        abort_match = {
+            CONF_COUNTRY: config[CONF_COUNTRY],
+            CONF_EXCLUDES: config[CONF_EXCLUDES],
+            CONF_OFFSET: config[CONF_OFFSET],
+            CONF_WORKDAYS: config[CONF_WORKDAYS],
+            CONF_ADD_HOLIDAYS: config[CONF_ADD_HOLIDAYS],
+            CONF_REMOVE_HOLIDAYS: config[CONF_REMOVE_HOLIDAYS],
+            CONF_PROVINCE: config.get(CONF_PROVINCE),
+        }
+        new_config = config.copy()
+        new_config[CONF_PROVINCE] = config.get(CONF_PROVINCE)
+
+        self._async_abort_entries_match(abort_match)
+        return await self.async_step_options(user_input=new_config)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self.data = user_input
+            return await self.async_step_options()
+        return self.async_show_form(
+            step_id="user",
+            data_schema=DATA_SCHEMA_SETUP,
+            errors=errors,
+        )
+
+    async def async_step_options(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle remaining flow."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            combined_input: dict[str, Any] = {**self.data, **user_input}
+            if combined_input[CONF_PROVINCE] == "none":
```

Comment:
It might be possible to hide the "province" if provinces are not available and use:
```suggestion
            if combined_input.get(CONF_PROVINCE, NONE_SENTINEL) == NONE_SENTINEL:
```